lib/aws-manager.js

```diff
@@ -81,142 +82,164 @@ class AwsManager {
     let pricingStartDate = new Date();
     pricingStartDate.setMinutes(pricingStartDate.getMinutes() - 30);
 
-    let livingInstances;
-    let livingSpotRequests;
-    let deadInstances;
-    let deadSpotRequests = {};
-    let allPricingHistory;
-    let availableAZ;
-
-    try {
-      livingInstances = await this.ec2.describeInstances({
-        Filters: [
-          {
-            Name: 'key-name',
-            Values: [this.keyPrefix + '*'],
-          },
-          {
-            Name: 'instance-state-name',
-            Values: ['running', 'pending'],
-          },
-        ],
-      });
-      debug('fetched living instance state');
+    this.__previousApiState = this.__apiState;
+    let apiState = {
+      instances: [],
+      requests: [],
+    };
+    let deadState = {
+      instances: [],
+      requests: [],
+    };
 
-      livingSpotRequests = await this.ec2.describeSpotInstanceRequests({
-        Filters: [
-          {
-            Name: 'launch.key-name',
-            Values: [this.keyPrefix + '*'],
-          }, {
-            Name: 'state',
-            Values: ['open'],
-          },
-        ],
-      });
-      debug('fetched living spot request state');
+    let stalledKills = [];
+    let availableAZ = {};
+    let allPricingHistory = {};
+
+    // It would be better to do it by classification than by region for
+    // better concurrency, but this is easier and not too slow considering
+    // the 75s iteration frequency
+    _.forEach(this.ec2, async (obj, region) => {
+      let response = await Promise.all([
```

Comment:
To makes this more readable I strongly propose we unpack... We can do this in two ways:
```js
let [
  livingInstances,
  livingSpotRequests,
  deadInstances,
  deadSpotRequests,
  availabilityZones,
  spotPriceHistory,
] = await Promise.all([
  ....
])
```

You can indeed make your life a tiny bit simpler with:

```js
let [
  livingInstanceReservations,
  livingSpotRequests,
  deadInstances,
  deadSpotRequests,
  availabilityZones,
  spotPriceHistory,
] = await Promise.all([
        // Living instances
        obj.describeInstances({
          Filters: [
            {
              Name: 'key-name',
              Values: [this.keyPrefix + '*'],
            },
            {
              Name: 'instance-state-name',
              Values: ['running', 'pending'],
            },
          ],
        }).promise().then(_.property('data.Reservations')),
]);
```

Or if you want to make your life very simple you can do:
```js
let [
  livingInstances,
  livingSpotRequests,
  deadInstances,
  deadSpotRequests,
  availabilityZones,
  spotPriceHistory,
] = await Promise.all([
        // Living instances
        obj.describeInstances({
          Filters: [
            {
              Name: 'key-name',
              Values: [this.keyPrefix + '*'],
            },
            {
              Name: 'instance-state-name',
              Values: ['running', 'pending'],
            },
          ],
        }).promise()
        .then(_.property('data.Reservations'))
        .then(r => _.map(r, _.property('Instances')))
        .then(_.flatten),
])
```

I think the last one might be over kill, but using `.then(_.property('data.Reservations'))` is reasonably sane....